mooncake-store/src/master_service.cpp

```diff
@@ -507,6 +509,46 @@ auto MasterService::PutEnd(const UUID& client_id, const std::string& key,
     return {};
 }
 
+auto MasterService::AddReplica(const UUID& client_id, const std::string& key,
+                               Replica& replica)
+    -> tl::expected<void, ErrorCode> {
+    MetadataAccessor accessor(this, key);
+    if (!accessor.Exists()) {
+        LOG(ERROR) << "key=" << key << ", error=object_not_found";
+        return tl::make_unexpected(ErrorCode::OBJECT_NOT_FOUND);
+    }
+    auto& metadata = accessor.Get();
+    if (client_id != metadata.client_id) {
+        LOG(ERROR) << "Illegal client " << client_id << " to PutEnd key " << key
+                   << ", was PutStart-ed by " << metadata.client_id;
+        return tl::make_unexpected(ErrorCode::ILLEGAL_CLIENT);
+    }
+    if (replica.type() != ReplicaType::LOCAL_DISK) {
+        LOG(ERROR) << "Invalid replica type: " << replica.type()
+                   << ". Expected ReplicaType::LOCAL_DISK.";
+        return tl::make_unexpected(ErrorCode::INVALID_REPLICA);
+    }
+    bool update = false;
+    for (size_t i = 0; i < metadata.replicas.size(); ++i) {
+        if (metadata.replicas[i].type() == ReplicaType::LOCAL_DISK) {
```

Comment:
After `update = true` is set here, why not break out of the loop? If there are two local disk replicas, both replicas would be updated to use the same client’s replica.

Mooncake currently supports multiple memory replicas, and our ongoing RL rollout work will also involve multi-replica scenarios. Because of this, I think we need to consider correctness under multi-replica conditions when implementing this.

If a KV has multiple memory replicas distributed across different clients, they may all call the `addReplica` interface. We need to take this situation into account.

In addition, regarding the code `metadata.replicas[i].type() == ReplicaType::LOCAL_DISK`, I wasn’t able to find where `metadata.replicas` is set to `LOCAL_DISK` during the `put` operation.